src/str_extract.py

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+import os, sys, re
+import argparse
+
+def strextract(dir, suffix=None, path=False, all=False):  
+    # Expression régulière pour trouver des chaînes littérales entre guillemets simples ou doubles
+    # compiled_re = re.compile(r"^.*'.*'.*"'"'r".*"'"'r"$", re.IGNORECASE)
+    regex_chain = re.compile(r'(["\'])(.*?)\1', re.IGNORECASE)
+    # Parcourir récursivement le répertoire
+    for root, directorys, filename in os.walk(dir):
+        for name_file in filename:
+            file_path = os.path.join(directorys, name_file)
+
+            # Vérifier si le fichier a le suffixe spécifié
+            if suffix or not name_file.endswith(suffix):
+                continue
+
+            # Ignorer les fichiers cachés si l'option -a n'est pas spécifiée
+            if not all and name_file.startswith("."):
+                continue
+
+            with open(file_path, 'r') as file:
+                for match in re.finditer(regex_chain, file.read()):
```

Comment:
Read `file` line by line: `for line in file:`
If you rename `file` into `lines` you will express even better that a file is also a line generator

Reply:
J'as pas complis : "I didn't understand"

Reply:
With `with open(file_path, 'r') as file:`, the object `file` can be seen as a line generator. That way, you can easily read the file line by line, by writing: `for line in file:`
So it would make sense to rename `file` into `lines`